backend/routes/shops.js

```diff
@@ -885,8 +886,12 @@ module.exports = function (router) {
 
         // Update the public and dataUrl in the shop's config to
         // reflect a possible change to the hostname.
-        additionalOpts.publicUrl = getShopPublicUrl(req.shop, netConfig)
-        additionalOpts.dataUrl = getShopDataUrl(req.shop, netConfig)
+        if (!req.body.publicUrl) {
```

Comment:
Why is this check necessary? 
If the admin FE sends a hostname, shouldn't we always update the public and data URL?

I'm thinking of a scenario where:
1. Config has an existing hostname, publicUrl and dataUrl.
2. Merchant updates hostname and saves config
3. Admin FE call BE with hostname, publicUrl and dataUrl set in req.body
=> that would result in hostname getting updated in the config but not dataUrl and publicUrl, which would then cause later on failures in the back-end to fetch data.

Reply:
The idea was that if the FE sends publicUrl or daraUrl, those values should be trusted. I needed to override them when doing testing locally so needed this change. Right now the only place publicUrl and dataUrl get sent from the FE is from a super-user page so this should affect regular merchants

Reply:
Sounds good. Perhaps add a comment so we don't get confused in the future when looking at this code.